File: typescript/design-as-you-code/CarKey.ts
import Car from "./Car";

export default class CarKey {
    constructor(private car: Car) {}

    authorizedFor(car: Car): boolean {
        return this.car === car;
    }

    insertAndUnlock() {
        this.car.unlock(this);
    }

    pressUnlock() {
        this.car.unlock(this);
    }
}

File: python/src/domain/carKey.py
class CarKey:
    def __init__(self, car):
        self._car = car

    def authorizedFor(self, car):
        return self._car == car

    def insertAndUnlock(self):
        self._car.unlock(self)

    def pressUnlock(self):
        self._car.unlock(self)